### PR TITLE
QRS-31, QRS-41, QRS-42, increase performance of zoom view loading and fix Zoom mode quits by itself

### PR DESCRIPTION
**Pull request by VitalInnovator**
Used cached data to improve loading speed of zoom view

### CODE REVIEW

**Comment by roszcz** [comment]
@VitalInnovator what was the reason the zoom was quitting? Please provide some context

**Comment by VitalInnovator** [comment]
@roszcz The reason was that when switching quickly, the zoom view didn't have time to update, and when pressed again, it tried to perform actions on a record that had already been removed from the list of available records (recordsToDisplay).

File: next/src/modules/dashboardPage/pages/DashboardPage.tsx [hunk 1]
```diff
@@ -41,11 +42,15 @@ const DashboardPage = () => {
   const { status } = useSession();
   const loading = status === 'loading';
 
+  const queryClient = useQueryClient();
+
+  const getCachedFragment = (id: string) =>
+    queryClient.getQueryData<EcgFragment>(['record', id]);
```
Comment (roszcz) [inline next/src/modules/dashboardPage/pages/DashboardPage.tsx:48]: I'm trying to learn more about react and next, so can you give me some more context about how this caching works? Just a comment here is enough

**Review by roszcz — APPROVED** [review]

File: next/src/modules/dashboardPage/pages/DashboardPage.tsx (hunk 1 above)
Comment (VitalInnovator) [reply next/src/modules/dashboardPage/pages/DashboardPage.tsx:48]: We are using [ReactQuery](https://tanstack.com/query/latest/docs/react/overview) - a fantastic library for handling some of our requests.
It provides an excellent caching and cache revalidation mechanism.
The basic principle is very simple. We pass a key and a request function - after the request is completed, the cache will be available using the key provided earlier.
In this specific case, we are retrieving a cached fragment that was fetched in another component (chart).

**Pull request merged**